internal/services/compute/managed_disk_resource.go

```diff
@@ -1000,3 +1002,27 @@ func resourceManagedDiskDelete(d *pluginsdk.ResourceData, meta interface{}) erro
 
 	return nil
 }
+
+// shutDownOnResize implements live resize restrictions according to https://docs.microsoft.com/en-us/azure/virtual-machines/linux/expand-disks#expand-without-downtime
+func shutDownOnResize(disk *disks.Disk, oldSizeGB, newSizeGB int) bool {
+	// OS disks can't be expanded without downtime.
+	if *disk.Properties.OsType != "" {
+		return true
+	}
+	// Disks smaller than 4 TiB can't be expanded to 4 TiB or larger without downtime.
+	if oldSizeGB < 4096 && newSizeGB >= 4096 {
+		return true
+	}
+	// Only  Premium SSD v1 and Standard SSD disks support live resize
+	for _, diskType := range []disks.DiskStorageAccountTypes{
+		disks.DiskStorageAccountTypesPremiumLRS,
+		disks.DiskStorageAccountTypesPremiumZRS,
+		disks.DiskStorageAccountTypesStandardSSDLRS,
+		disks.DiskStorageAccountTypesStandardSSDZRS,
+	} {
+		if *disk.Sku.Name == diskType {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
based on the documentation for both Linux and Windows, we also need to check the VM that we're attached too, including that it supports Expand without Downtime (https://learn.microsoft.com/azure/virtual-machines/linux/expand-disks?tabs=azure-cli%2Cubuntu#expand-without-downtime / https://learn.microsoft.com/azure/virtual-machines/windows/expand-os-disk#expand-without-downtime) - so we'll need to first retrieve the Managed Disk, confirm it's a Virtual Machine, and then retrieve the Virtual Machine to confirm it supports this (else we'll need to fallback as we're doing here)